### PR TITLE
Add orca gbw

### PR DESCRIPTION
**Pull request by briling**
* [x] check .gbw reader @ 4.0.0 and 4.2.1
* [x] ? manual correction for 3d @ def2svp etc -- is the order version dependent?

### CODE REVIEW

File: qstack/orcaio.py [hunk 1]
```diff
@@ -38,8 +42,115 @@ def read_density(mol, basename, directory='./', version=500, openshell=False, re
...
+        coefficients_ab, energies_ab, occupations_ab = read_mos()
+        ls = read_basis()
+        return coefficients_ab, energies_ab, occupations_ab, ls
+
+
+def reorder_coeff_inplace(c_full, mol, reorder_dest='pyscf', ls_from_orca=None):
+    def _reorder_coeff(c):
+        # In ORCA, at least def2-SVP and def2-TZVP for 3d metals
+        # are not sorted wrt angular momenta
+        # TODO add the fix to read_density()   #if gto.basis._format_basis_name(mol.basis)=='def2tzvp':
```
Comment (YAY-C) [inline qstack/orcaio.py:123]: I think you did that (and can remove the `TODO`)!
_unless you wanted to do something more than throwing a `warn()`_

File: qstack/orcaio.py [hunk 2]
```diff
@@ -38,8 +42,115 @@ def read_density(mol, basename, directory='./', version=500, openshell=False, re
...
+                indices = np.array([j for i in indices for j in i[2]])
+                ao_limits = mol.offset_ao_by_atom()[iat][2:]
+                atom_slice = np.s_[ao_limits[0]:ao_limits[1]]
+                indices_full[atom_slice] = indices[:] + ao_limits[0]
+            for i in range(len(c)):
+                c[:,i] = c[indices_full,i]
+
+        for i in range(len(c)):
+            c[:,i] = reorder_ao(mol, c[:,i], src='orca', dest=reorder_dest)
+    [_reorder_coeff(c_full[i]) for i in range(c_full.shape[0])]
```
[the remark this answers is not in the document]
Comment (briling) [reply qstack/orcaio.py:142]: I did write it, just wanted to have a one liner but you're right it looks weird

**Comment by briling** [comment]
> Maybe it's worth to specifying that we only deal with version 400 and 421 (in the doc or so)?

Orca  versions `4.0.0` and `4.2.1` are available on `lc2` and `5.0.4` is on `lc3`. We may need to access some computations obtained with all of them (recently I had to read some old `.scfp` file by `4.0.0`). As you see from the code, the `.scfp`/`.scfr` format has changed between `4.0` and `4.2`, and in `5.0` they changed it to `.densities`. I haven't implemented readers for other versions because don't think we need it & don't want to install them but this can be done.

What would you say in the doc?

Actually I need to check if the `.gbw` format is the same for all the versions:
* [ ] check `.gbw` reader @ `4.0.0` and `4.2.1`

> unless you wanted to do something more than throwing a warn()

I changed one of the warnings to error because it doesn't make sense to change AO order from orca to pyscf or whatever when the original order is wrong. Still, one can load the file without reordering, correct the 3d elements or anything manually and then reorder for pyscf. 

Alternatively, I can just add a manual correction for 3d elements @ def2-svp and others.

Also I was thinking about trying to find a `.gbw` file in the same directory and using the basis set from it,
but it is weird because then you can read orbitals and then you don't need the density files.

What do you think?

**Comment by briling** [comment]
@YAY-C do you think I can merge?

**Review by YAY-C — APPROVED** [review]

**Comment by YAY-C** [comment]
Sorry, I just saw the message!
But yes, I think it's ready for merging :+1:

**Pull request merged**